test/integration/skvclient.py

```diff
@@ -300,6 +300,18 @@ def make_record(self, **dataFields):
                 rec.fields.__dict__[dname] = None
         return rec
 
+    # For use in creating key records for a prefix query. The difference is fields that are not set
+    # are not set in excludedFields and are not set to None
+    def make_prefix_record(self, **dataFields):
+        rec = Record(self.name, self.version)
+        for i,field in enumerate(self.fields):
+            dname = field.name.decode('ascii')
+            if dname in dataFields:
+                fieldValue = dataFields.get(dname)
+                rec._posFields.append(fieldValue)
+                rec.fields.__dict__[dname] = fieldValue
```

Comment:
i think in the else part we should break here to make sure we don't set any other fields. Maybe throw after if there are some dataFields left that we didn't use (just check if len(dataFields) == len(rec._posFields))